src/test/controllerscriptenginelegacy_test.cpp

```diff
@@ -57,13 +73,84 @@ class ControllerScriptEngineLegacyTest : public ControllerScriptEngineLegacy, pu
+        // but it does a lot of local disk / settings setup.
+        auto pChannelHandleFactory = std::make_shared<ChannelHandleFactory>();
+        m_pEffectsManager = std::make_shared<EffectsManager>(config(), pChannelHandleFactory);
+        m_pEngine = std::make_shared<EngineMixer>(
+                config(),
+                "[Master]",
+                m_pEffectsManager.get(),
+                pChannelHandleFactory,
+                true);
+        m_pSoundManager = std::make_shared<SoundManager>(config(), m_pEngine.get());
+        m_pControlIndicatorTimer = std::make_shared<mixxx::ControlIndicatorTimer>(nullptr);
+        m_pEngine->registerNonEngineChannelSoundIO(gsl::make_not_null(m_pSoundManager.get()));
+                
+        CoverArtCache::createInstance();
+
+        m_pPlayerManager = std::make_shared<PlayerManager>(config(),
+                m_pSoundManager.get(),
+                m_pEffectsManager.get(),
+                m_pEngine.get());
+
+        m_pPlayerManager->addConfiguredDecks();
+        m_pPlayerManager->addSampler();
+        PlayerInfo::create();
+        m_pEffectsManager->setup();
+
+        const auto dbConnection = mixxx::DbConnectionPooled(dbConnectionPooler());
+        if (!MixxxDb::initDatabaseSchema(dbConnection)) {
+            exit(1);
+        }
+
+        m_pTrackCollectionManager = std::make_shared<TrackCollectionManager>(
+                nullptr,
+                config(),
+                dbConnectionPooler(),
+                [](Track* pTrack) { delete pTrack; });
+
+        m_pRecordingManager = std::make_shared<RecordingManager>(config(), m_pEngine.get());
+        m_pLibrary = std::make_shared<Library>(
+                nullptr,
+                config(),
+                dbConnectionPooler(),
+                m_pTrackCollectionManager.get(),
+                m_pPlayerManager.get(),
+                m_pRecordingManager.get());
+
+        m_pPlayerManager->bindToLibrary(m_pLibrary.get());
+        ControllerScriptEngineBase::registerPlayerManager(m_pPlayerManager);
+        ControllerScriptEngineBase::registerTrackCollectionManager(m_pTrackCollectionManager);
     }
 
+    // Helper to get or add a track by location, like in PlayerManagerTest
+    TrackPointer getOrAddTrackByLocation(const QString& trackLocation) {
+        return m_pTrackCollectionManager->getOrAddTrack(
+                TrackRef::fromFilePath(trackLocation));
+    }
     void TearDown() override {
         mixxx::Time::setTestMode(false);
 #ifdef MIXXX_USE_QML
         m_rootItems.clear();
 #endif
+        CoverArtCache::destroy();
+        ControllerScriptEngineBase::registerPlayerManager(nullptr);
+        ControllerScriptEngineBase::registerTrackCollectionManager(nullptr);
+    }
+
+    ~ControllerScriptEngineLegacyTest() {
+        // Reset in the correct order to avoid singleton destruction issues
+        m_pSoundManager.reset();
```

Comment:
I'm not sure if you can have both TearDown and a dtor in a test. I guess if so teardown gets called first?